### DISCUSSION TITLE
xDrip Plus Build Tools

### DISCUSSION THREAD

**Discussion by Miquest** [NightscoutFoundation/xDrip#2926] [Q&A] [answered]
Hi,
which build tools do I have to use for xDrip?
I tried to build xDrip APK (I had to patch it myself) with Gradle 7.2, current JDK/NDK.and Android Studio Flamingo (current Update). 
But Gradle 7.2 doesn't include the maven Plugin and the JDK doesn't work as well.

Which versions of Gradle, JDK/NDK do I have to use for xDrip Plus Development?

**Comment by Navid200** [comment]
Bumblebee
JDK 11

**Reply by walternautic** [reply]
Hi,
I tried to start with some xDrip development but failed to setup my Android Studio environment so that xDrip builds successfully.

It seems, I am not the only one with problems (see also #3066), so I want to collect all infos that helps to setup a valid xDrip development environment.

So far, I found the following info:
* Android Studio Bumblebee (download version 2021.1.1 at https://developer.android.com/studio/archive)
* JDK 11 (can be selected in the Android Studio settings) - is there a preferred JDK provider? (Eclipse Temurin, Oracle etc.)
* Lombok plugin - I could not find it and found some StackOverflow threads about it: https://stackoverflow.com/questions/70900954/cannot-find-lombok-plugin-in-android-studio-2021-1-1-build-of-2022. Anyhow I could not find any version that works with Bumblebee. Are there any tricks how to install the Lombok plugin?
* Gradle plugin must not be upgraded - but which version should be used? Which Gradle version can be used?

https://github.com/NightscoutFoundation/xDrip/blob/master/CONTRIBUTING.md also doesn't give any hint how to setup a working environment.

Thanks for any hints!
Walter

**Reply by Navid200** [reply]
@walternautic 
I am using this: https://github.com/mplushnikov/lombok-intellij-plugin/releases/tag/releasebuild_0.34.1-2018.1

2019.1


There is a thought of including it in the project tree.  But, it's not decided yet.

**Reply by jamorham** [reply]
I think later versions of Android studio may have the lombok issue fixed as it will be available via the normal plugin installation route again but I haven't tested this version (currently in canary release). I use bumblebee version and android studio bundled jdk 11 which I believe is the default for this edition of android studio. I also use one of the third-party extracted lombok plugins suitable for bumblebee. Don't change the gradle version from what is in the source tree. The fact that later versions of android studio have problems compiling software which works fine on previous versions is a regression in my opinion and these have been introduced in every release of android studio which is why I'm not that keen on upgrading the application.

**Comment by Navid200** [comment, marked as answer]
We are now using Giraffe.
The default SDK that comes with it is what we use.
Lombok and more detail is here: 
https://github.com/NightscoutFoundation/xDrip/blob/master/Documentation/Quick_Start.md
